test/one/profiler/test/TestProcess.java

```diff
@@ -85,6 +85,13 @@ public String[] inputs() {
         return this.inputs;
     }
 
+    private static boolean needsDynamicAgentLoadingFlag() {
+        String version = System.getProperty("java.version");
+        String[] versionParts = version.split("\\.");
+        int majorVersion = Integer.parseInt(versionParts[0]);
+        return majorVersion >= 21;
+    }
```

Comment:
Target JDK is not necessary the same as the host JDK.

Reply:
oh, you're right